File: dev-project-backend/backend-core/src/main/java/dev/remitano/core/service/VideoService.java
package dev.remitano.core.service;

import dev.remitano.core.models.Video;
import dev.remitano.infrastructure.dto.request.SharingQueueDto;
import dev.remitano.infrastructure.enumeration.VoteType;
import org.springframework.data.domain.Page;

/**
 * @author anhdx
 */
public interface VideoService {

    /**
     * get all video
     * @return
     */
    Page<Video> getAllVideo(int page, int pageSize);

    /**
     * push to queue to handle async share video
     * @param url
     */
    void pushShareVideo(String url);

    /**
     * shareVideo
     * @param dto
     * @return
     */
    Video shareVideo(SharingQueueDto dto);

    /**
     * vote video
     * @param type
     * @return
     */
    Video voteVideo(Long videoId, VoteType type);

}

File: dev-project-frontend/src/app/service/video.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {HttpService} from '../core/service/http.service';

@Injectable()
export class VideoService extends HttpService {

  constructor(httpClient: HttpClient) {
    super(httpClient);
  }

  public getAllVideo(page: number, pageSize: number): Promise<any> {
    return this.getAPI('/api/videos?page=' + page + "&pageSize=" + pageSize);
  }

  public voteVideo(id: number, type: number): Promise<any> {
    return this.putAPI('/api/videos/' + id + "/vote?type=" + type, null);
  }

  public shareVideo(data: any): Promise<any> {
    return this.postAPI('/api/videos', data);
  }

}
